conjure-go-client/httpclient/client_builder.go

```diff
@@ -100,8 +101,11 @@ func NewClient(params ...ClientParam) (Client, error) {
 	}
 
 	if b.maxRetries == 0 {
-		b.maxRetries = 2 * len(b.uris)
+		if b.uris != nil {
+			b.maxRetries = 2 * len(b.uris.CurrentStringSlice())
```

Comment:
I know MaxRetries isn't a refreshable yet, but it would be great to move this to within `Do` so it can change if the list of uris changes

Reply:
Thats a good point, `MaxRetries` should in theory be easy to make refreshable (in a similar fashion to uris) I'll make it next on my list to do and move this logic along with it